packages/react-charting/src/components/HorizontalBarChart/HorizontalBarChart.base.tsx

```diff
@@ -288,19 +295,47 @@ export class HorizontalBarChartBase extends React.Component<IHorizontalBarChartP
     );
   }
 
+  // Adjusts the margin percentage, based on the width of the svg (inversly propotional)
+  private _adjustBarSpacing(): void {
+    const svgWidth = this.barChartSvgRef.current?.getBoundingClientRect().width;
+    if (svgWidth) {
+      // [0,1] Controls the spacing between the bars
+      const BAR_SPACING_FACTOR = 1;
+      // The barspacing is provided in percentage terms, so changes inversely to the width of the svg
+      const currentBarSpacing = clamp((200 / svgWidth) * BAR_SPACING_FACTOR, 1, 0.2);
```

Comment:
should you be using the total width of the container instead of this number?

Reply:
So here we're trying to define what percentage of the total svg's width should be available as a space b/w 2 consecutive bars.

Points Considered:
- In case of svg, we cannot specify this spacing in pixels. (since svg is scalable). Hence percentage is needed
- Now having a constant percentage of space for margins, also doesn't work well, since if the width of the svg is very low the absolute px value of the svg would be too less and v.v
- We want the percentage for margins to scale inversely to the width of the svg, to get a near consistent absolute pixel value for the margin (that is what we'd ideally need)
- But again we must clamp the value, since we would not want the % to decrease too much and become nearly 0 in case of a very wide svg, and similarly become too high and approach "infinity" in case of a very small svg. 

The value 200, is obtained after fine-tuning based on the margin that's required for the focus indicator to be visible w/o being cropped.

_Examples_
1. If svgWidth=200. 1% of the space would be used for each margin, which is 2px. That matches the width of the focus indicator.
2. If svgWidth=100. 2% of the space would be used for each margin, which is 2px. That also matches the width of the focus indicator.

The alternative to this solution is using stroke on the bars and using the <use> tag to change the stacking order OR use d3.raise() function that we discussed a while back.

Reply:
ok. Great. Add this explanation in the code where you defined the 200 value.